### PR TITLE
Upgrading parquet-format to 4.0.0

### LINKED ISSUE #978: Support for parquet-format latest version
Current parquet is using pinned version of parquet-format **~2.6.1** This is limiting ability of using parquet format of 2.6.0

parquet-format crate released new version which is 4.0 this conforms to parquet format of 2.8.0. I think it's safe to use the latest version so that we can leverage newer support of the parquet.

### PR DESCRIPTION
**Pull request by kesavkolla**
Closes #978 .

# Rationale for this change
Support for newer parquet format
 

# What changes are included in this PR?
Added extra fields that are part of metada structures

# Are there any user-facing changes?
No

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @kesavkolla  - I started the CI test runs and if they pass I plan to merge this in (and it will probably appear in arrow 7.0.0, release in ~ February)

File: parquet/src/basic.rs [hunk 1]
```diff
@@ -250,6 +250,15 @@ pub enum Encoding {
...
     RLE_DICTIONARY,
+
+    /// Encoding for floating-point data.
+    ///
+    /// K byte-streams are created where K is the size in bytes of the data type.
+    /// The individual bytes of an FP value are scattered to the corresponding stream and
+    /// the streams are concatenated.
+    /// This itself does not reduce the size of the data but can lead to better compression
+    /// afterwards.
+    BYTE_STREAM_SPLIT
```
Comment (Jimexist) [inline parquet/src/basic.rs:261]:
```suggestion
    BYTE_STREAM_SPLIT,
```

**Comment by alamb** [comment]
FYI @nevi-me

**Comment by alamb** [comment]
Thanks again @kesavkolla !

**Pull request merged**